pyiron_base/generic/dynamic.py

```diff
@@ -0,0 +1,60 @@
+import os
+import json
+from pyiron_base.state import state
+
+
+def _get_class_path_lst():
+    path_lst = [
+        os.path.abspath(os.path.expanduser(os.path.join(p, "template")))
+        for p in state.settings.resource_paths
+        if "template" in os.listdir(p)
+    ]
+    class_path_lst = []
+    for path in path_lst:
+        class_path_lst += [
+            cp
+            for cp, files in [
+                [cp, os.listdir(cp)]
+                for cp in [os.path.join(path, c) for c in os.listdir(path)]
+            ]
+            if "script.py" in files and "input.json" in files
+        ]
+    return class_path_lst
+
+
+def _load_input(path):
+    with open(os.path.join(path, "input.json")) as f:
+        return json.load(f)
+
+
+def _init_constructor(class_name, script_path, input_dict):
+    def __init__(self, project, job_name):
+        super(self.__class__, self).__init__(project, job_name)
+        self.__name__ = class_name
+        self.input.update(input_dict)
+        self.script_path = script_path
+
+    return __init__
+
+
+def _class_constructor(cp):
+    from pyiron_base.job.script import ScriptJob
```

Comment:
The local import inside the function is not so nice - but otherwise I always got infinite import loops.